src/operations/StandardOperation.cpp

```diff
@@ -467,4 +467,94 @@ void StandardOperation::dumpOpenQASMTeleportation(
   of << "teleport " << qreg[targets[0]].second << ", "
      << qreg[targets[1]].second << ", " << qreg[targets[2]].second << ";\n";
 }
+
+void StandardOperation::invert() {
+  switch (type) {
+  // self-inverting gates
+  case I:
+  case X:
+  case Y:
+  case Z:
+  case H:
+  case SWAP:
+  case ECR:
+    break;
+  // gates where we just update parameters
+  case GPhase:
+  case Phase:
+  case RX:
+  case RY:
+  case RZ:
+  case RXX:
+  case RYY:
+  case RZZ:
+  case RZX:
+    parameter[0] = -parameter[0];
+    break;
+  case U2:
+    std::swap(parameter[0], parameter[1]);
+    parameter[0] = -parameter[0] + PI;
+    parameter[1] = -parameter[1] - PI;
+    break;
+  case U3:
+    parameter[0] = -parameter[0];
+    parameter[1] = -parameter[1];
+    parameter[2] = -parameter[2];
+    std::swap(parameter[1], parameter[2]);
+    break;
+  case DCX:
+    std::swap(parameter[0], parameter[1]);
```

Comment:
It's not about swapping the parameters here, but really about swapping the targets, i.e., the inverse of DCX(0, 1) is DCX(1, 0).
Could be as easy as
```suggestion
    std::swap(targets[0], targets[1]);
```